packages/alfa-rules/test/sia-r8/rule.spec.tsx

```diff
@@ -197,3 +197,159 @@ test("evaluate() is inapplicable for an element which is not displayed", async (
 
   t.deepEqual(await evaluate(R8, { document }), [inapplicable(R8)]);
 });
+
+test(`evaluate() fails an input element with type=password which is disabled 
+    and without accessible name`, async (t) => {
+  const target = <input type="password" disabled/>;
+
+  const document = h.document([target]);
+
+  t.deepEqual(await evaluate(R8, { document }), [
+    failed(R8, target, {
+      1: Outcomes.InputElementWithNoAriaRoleHasNoName("password"),
+    }),
+  ]);
+});
+
+test("evaluate() passes an input element with type=password and implicit label", async (t) => {
+  const target = <input  type="password"/>;
+
+  const label = (
+    <label>
+      first name
```

Comment:
```suggestion
      password
```
(even if not important for the rule, let's make the examples look good or we'll be wondering about them in a couple of years)